demos/mqtt/mqtt_demo_lightweight/mqtt_demo_lightweight.c

```diff
@@ -849,34 +876,63 @@ int main( int argc,
-
-            /* Process incoming packet from the broker. After sending the subscribe, the
-             * client may receive a publish before it receives a subscribe ack. Therefore,
-             * call generic incoming packet processing function. Since this demo is
-             * subscribing to the topic to which no one is publishing, probability of
-             * receiving Publish message before subscribe ack is zero; but application
-             * must be ready to receive any packet.  This demo uses the generic packet
-             * processing function everywhere to highlight this fact. */
-            mqttProcessIncomingPacket( &networkContext, &fixedBuffer );
+            /**************************** Subscribe, Re-subscribe, and Keep-Alive ******************************/
+
+            /* Initialize retry attempts and interval. */
+            Transport_ReconnectParamsReset( &reconnectParams );
+
+            do
+            {
+                /* The client is now connected to the broker. Subscribe to the topic
+                 * as specified in MQTT_EXAMPLE_TOPIC at the top of this file by sending a
+                 * subscribe packet then waiting for a subscribe acknowledgment (SUBACK).
+                 * This client will then publish to the same topic it subscribed to, so it
+                 * will expect all the messages it sends to the broker to be sent back to it
+                 * from the broker. This demo uses QOS0 in subscribe, therefore, the Publish
+                 * messages received from the broker will have QOS0. */
+                /* Subscribe and SUBACK */
+                LogInfo( ( "Attempt to subscribe to the MQTT topic %s\r\n", MQTT_EXAMPLE_TOPIC ) );
+                mqttSubscribeToTopic( &networkContext, &fixedBuffer );
+
+                /* Since subscribe is a control packet, record the last control packet sent
+                 * timestamp. This timestamp will be used to determine if it is necessary to
+                 * send a PINGREQ packet. */
+                returnStatus = clock_gettime( CLOCK_MONOTONIC, &currentTimeStamp );
+                assert( returnStatus == 0 );
+                lastControlPacketSentTimeStamp = currentTimeStamp.tv_sec;
+
+                /* Process incoming packet from the broker. After sending the subscribe, the
+                 * client may receive a publish before it receives a subscribe ack. Therefore,
+                 * call generic incoming packet processing function. Since this demo is
+                 * subscribing to the topic to which no one is publishing, probability of
+                 * receiving Publish message before subscribe ack is zero; but application
+                 * must be ready to receive any packet.  This demo uses the generic packet
+                 * processing function everywhere to highlight this fact. */
+                mqttProcessIncomingPacket( &networkContext, &fixedBuffer );
+
+                if( globalSubscribeStatus == false )
```

Comment:
Please add a comment here to describe the reason for checking the `globalSubscribeStatus` value. We want to be very descriptive about the logic in the demo code to help the customer

Reply:
Also, add a comment within the block about why the ping request is sent to the broker